### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by DavidDeSimone**
As we build up momentum and features, I think it's good to update the readme with information based on how to build, and why the user should use emacs-ng. I've partially based this on remacs README.md with some tweaks. We we add new features like git2-rs, and stabilize features like webrender, we will update the README showing the power behind those features too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,158 @@
...
+- [emacs-ng](#emacs-ng)
+    - [Motivation](#motivation)
+- [Why Emacsng](#why-emacsng)
+    - [Why JavaScript](#why-javascript)
+        - [Performance](#performance)
+- [Getting Started](#getting-started)
+    - [Requirements](#requirements)
+    - [Building emacsng](#building-emacsng)
+    - [Running emacsng](#running-emacsng)
+    - [Features](#features)
```
Comment (brotzeit) [inline README.md:15]: Should be one level up right ?

Comment (DavidDeSimone) [reply README.md]: Updated

**Comment by brotzeit** [comment]
I had to download the submodules, can't find it in the readme though.

**Comment by DavidDeSimone** [comment]
> I had to download the submodules, can't find it in the readme though.

Good Catch! Updated.

**Pull request merged**